modules/tinymce/src/core/main/ts/api/html/Schema.ts

```diff
@@ -67,7 +67,6 @@ export interface SchemaRegExpMap { [name: string]: RegExp }
 
 interface Schema {
   children: Record<string, SchemaMap>;
-  elements: Record<string, SchemaElement>;
```

Comment:
Oh dear. I forgot about this in the initial discussions around schema changes, but RTC relies on updating this record (we make sure this happens before it's used). cc @spocke we will need to have a discussion about how to deal with this change properly - maybe it means putting our alt-model schema patches into core inside one of the "if rtc" checks.

Reply:
Tbh I think we can revert this if needed. I'd need to double check why Neil removed it in the first place though (I think it was just since it was more of an internal attribute, but if that's the case `children` should also be removed)

Reply:
I think it was because `setValidElements` does `elements = {}`, meaning the exported reference is no longer used. Aside from making `schema.elements` out of date it's also a memory leak.

Reply:
Ahh right, that does ring a bell now that you mention it 👍

Reply:
Yes, putting it in a isRtc branch seems like the best option here. Then we don't need to expose it anymore. So who does that can that be added to this PR? I think it's just copy paste of the raw code from the rockfest repo to the schema. Just put it in a function like mutateRtcElements or something and run that in RTC mode.

Reply:
I'd prefer to not do that, as I already really dislike having a heap of branching in core. So instead I'd like to find a way to expose additional elements to be added.

Reply:
Hmm, then it will be more complex since it needs to expose reading it and writing to it so that can't easily be done without putting the reference out as it was before. I'm not sure I will have time to do any extensive work on this on the RTC side so rollback the change or mutate internally as Andy suggested is the two options I have in order to finish this within a week.

Reply:
I can think of 2 options that avoid the branching:
1. Revert so `elements` is exposed and then remove all existing properties instead of doing `elements = {}` (simplest so probably the better given time constraints)
2. Add a new `addValidElement(name, attrSpec?)` API. This is risky doing it quickly and could be confusing without a `removeValidElement` API, but it would work for what RTC needs I believe.

Reply:
I think this is the only place we mutate the elements record so 2 might work but as you say that is more work and not scheduled work.
https://github.com/tinymce/rockfest/blob/master/modules/rtc-plugin/src/main/re/model/ImportTinymceSchema.res#L21

I think doing 1 now is better and we can figure out how to do this proper later. Children should also really go though a api at some point.

Reply:
Sounds good 👍  I'll get that done first thing on Thursday then.

Reply:
Have logged https://github.com/tinymce/tinymce/pull/7562 to do option 1.